Frontend/src/pages/registration_edit/components/RegistrationEditor.jsx

```diff
@@ -1,35 +1,43 @@
 import { EventSelector } from '@thewca/wca-components'
-import React, { useEffect, useState } from 'react'
+import React, { useContext, useEffect, useState } from 'react'
 import { Button, Checkbox, TextArea } from 'semantic-ui-react'
-import { useHeldEvents } from '../../../api/helper/hooks/use_competition_info'
-import useCompetitorInfo from '../../../api/helper/hooks/use_competitor_info'
+import { CompetitionContext } from '../../../api/helper/context/competition_context'
 import { getSingleRegistration } from '../../../api/registration/get/get_registrations'
 import { updateRegistration } from '../../../api/registration/patch/update_registration'
 import { setMessage } from '../../../ui/events/messages'
 import LoadingMessage from '../../../ui/messages/loadingMessage'
 import styles from './editor.module.scss'
+import { useQuery } from '@tanstack/react-query'
+import getCompetitorInfo from '../../../api/user/get/get_user_info'
 
 export default function RegistrationEditor({ user_id, competition_id }) {
-  const { isLoading: eventsLoading, heldEvents } = useHeldEvents(competition_id)
-  const { isLoading: infoLoading, competitorInfo } = useCompetitorInfo(user_id)
-  const [registration, setRegistration] = useState({})
+  const { competitionInfo } = useContext(CompetitionContext)
   const [comment, setComment] = useState('')
   const [status, setStatus] = useState('')
   const [selectedEvents, setSelectedEvents] = useState([])
+  const [registration, setRegistration] = useState({})
+  const { data: serverRegistration } = useQuery({
+    queryKey: [competition_id, user_id],
+    queryFn: () => getSingleRegistration(user_id, competition_id),
+  })
+  const { isLoading, data: competitorInfo } = useQuery({
+    queryKey: ['info', user_id],
+    queryFn: () => getCompetitorInfo(user_id),
+    enabled: Boolean(serverRegistration),
```

Comment:
`useQuery` provides helpful things like `status` and `isSuccess`, which are designed for things like this (in place of `Boolean(serverRegistration)`).

But why does this query need to be disabled at all? Can't both queries run at once?

One potentially annoying/confusing thing is that `isLoading` from `useQuery` is either `loading`, `success`, or `error`. So if the query is disabled then it's neither a success nor an error so `isLoading` defaults to `true`... even though the query isn't loading.

Reply:
I removed the enabled, I needed it before refactoring something and forgot to remove it.